README.md

```diff
@@ -75,37 +75,44 @@ Usage: ruby ticketmatch.rb [options]
     -h, --help                       this message
 ```
 
-# Batch processing
+# Batch processing for puppet-agent and its components
 
-If you need to check all the components that comprise `puppet-agent` for a release, use
-`pa_matchbatch.sh`. It requires one argument: the revision (SHA or reference) of `puppet-agent` you
-care about. It will process `puppet-agent` and all its components that are tagged with a SHA.
 
-Component directories will be cloned from the path provided in their `component.json` file and the
-appropriate branch set. If a given component already exists, a Git fetch will be performed to get
-things up to date (see environment variables, below, for additional options).
+The `pa_matchbatch.sh` script clones a revision of puppet-agent and all of its components, then runs
+ticketmatch on everything; This is intended for use during puppet-agent releases.
 
-## How to run pa_matchbatch (one way)
+### How to run pa_matchbatch
 
-set your working directory to someplace (cannot be `puppet-agent`; can be its parent or empty)
+*`pa_matchbatch.sh` requires one argument,* A puppet-agent git reference (a branch name, a SHA, or
+a tag). For example, to run ticketmatch on puppet-agent's master branch and all of its components:
 
-```cd <a_directory>```
-
-invoke pa_matchbatch
+```sh
+./pa_matchbatch.sh <branch-name>
+```
 
-```path/pa_matchbatch.sh <the_revision>```
+*`pa_matchbatch.sh` clones puppet-agent and its components to the current working directory by default.*
+You can override the target directory by setting the `$WORKSPACE` environment variable to some other
+directory. You can run `pa_matchbatch.sh` any number of times on the same workspace; The repos will
+be fetched each time to check for new updates.
 
-## How to run pa_matchbatch (another way)
+*`pa_matchbatch.sh` assumes that only the Z versions of puppet-agent and its components will change
+during a release.* You will need to override this assumption for:
```

Comment:
The description of why versions might be incorrect was incomplete (the note about Z versions was the reason I started these edits), and this is quite important, so I moved it upward